source/camera/CullingManager.cpp

```diff
@@ -0,0 +1,292 @@
+    mDebug(false),
+    mFirstIter(false),
+    mCm(cameraManager),
+    mCullCreaturesFlag(false),
+    mCullTilesFlag(false)
+{
+    mMyplanes =(Ogre::Plane(0, 0, 1, 0));
+    // init the Ogre vector
+    for (unsigned int i = 0; i < 8; ++i)
+    {
+        mOgreVectorsArray[i].x = static_cast<Ogre::Real>(0.0);
+        mOgreVectorsArray[i].y = static_cast<Ogre::Real>(0.0);
+        mOgreVectorsArray[i].z = static_cast<Ogre::Real>(0.0);
+    }
+}
+
+void CullingManager::cullTiles()
+{
+    mOldWalk = mWalk;
+    mWalk.mVertices.mMyArray.clear();
+    for (int ii = 0 ; ii < 8 ; ++ii)
+        mWalk.mVertices.mMyArray.push_back(Vector3i(mOgreVectorsArray[ii]));
+
+    // create a slope -- a set of left and rigth path
+    mWalk.convex_hull();
+    mWalk.buildSlopes();
+
+    if(mDebug)
+    {
+        LogManager::getSingleton().logMessage(LogMessageLevel::NORMAL,__FILE__,__LINE__ ,mOldWalk.debug());
+        LogManager::getSingleton().logMessage(LogMessageLevel::NORMAL,__FILE__,__LINE__ ,mWalk.debug());
+        
+    }
+
+    // reset index pointers to the begging of collections
+    mOldWalk.prepareWalk();
+    mWalk.prepareWalk();
+
+    newBashAndSplashTiles(SHOW | HIDE);
+}
+
+void CullingManager::startCreatureCulling()
+{
+    mCullCreaturesFlag = true;
+}
+
+void CullingManager::startTileCulling()
+{
+    getIntersectionPoints();
+
+    mWalk.mVertices.mMyArray.clear();
+    for (int ii = 0 ; ii < 8 ; ++ii)
+        mWalk.mVertices.mMyArray.push_back(Vector3i(mOgreVectorsArray[ii]));
+    mWalk.convex_hull();
+    mWalk.buildSlopes();
+    mOldWalk = mWalk;
+    mOldWalk.prepareWalk();
+    mWalk.prepareWalk();
+    hideAllTiles();
+    newBashAndSplashTiles(SHOW);
+
+    mCullTilesFlag = true;
+}
+
+void CullingManager::startDebugging()
+{
+    mDebug = true;
+}
+
+void CullingManager::stopCreatureCulling()
+{
+    mCullCreaturesFlag = false;
+}
+
+void CullingManager::stopTileCulling()
+{
+    mCullTilesFlag = false;
+    mOldWalk = mWalk;
+    mWalk.mVertices.mMyArray.clear();
+    for (int ii = 0 ; ii < 8 ; ++ii)
+        mWalk.mVertices.mMyArray.push_back(Vector3i(mOgreVectorsArray[ii]));
+
+    // create a slope -- a set of left and rigth path
+    mWalk.convex_hull();
+    mWalk.buildSlopes();
+
+    if(mDebug)
+    {
+        LogManager::getSingleton().logMessage(LogMessageLevel::NORMAL,__FILE__,__LINE__ ,mOldWalk.debug());
+        LogManager::getSingleton().logMessage(LogMessageLevel::NORMAL,__FILE__,__LINE__ ,mWalk.debug());
+        
+    }
+
+    // reset index pointers to the begging of collections
+    mOldWalk.prepareWalk();
+    mWalk.prepareWalk();
+
+    newBashAndSplashTiles(HIDE);
+    showAllTiles();
+
+}
+
+void CullingManager::hideAllTiles(void)
+{
+    GameMap* gm = mCm->mGameMap;
+    if (!gm)
+        return;
+
+    for (int jj = 0; jj < gm->getMapSizeY() ; ++jj)
+    {
+        for (int ii = 0; ii < gm->getMapSizeX(); ++ii)
+        {
+            gm->getTile(ii,jj)->hide();
```

Comment:
it looks nicer and more abstract, we don't look how the hide and show methods are implemented .